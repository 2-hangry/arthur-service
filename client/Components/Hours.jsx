import React from "react";
import Day from "./DailyHours.jsx";
const divStyle = {
  color: "red",
  fontFamily: "sans-serif",
  fontSize: 14+'px'
};

const Hours = props => {
  return (
    <div>
      <h3 style={divStyle}>Hours</h3>
      <table>
        <tbody>
          {props.hours.open.map(day => {
            return <Day day={day} />;
          })}
        </tbody>
      </table>
    </div>
  );
};

export default Hours;
