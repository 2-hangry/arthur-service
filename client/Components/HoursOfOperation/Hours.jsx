import React from 'react';
import DailyHours from './DailyHours.jsx';
import styles from '../../styles.css.js';

const Hours = props => (
  <div>
    <p style={styles.sectionTitle}>Hours</p>
    <table>
      <tbody>
        {props.hours.open.map(day => <DailyHours day={day} />)}
      </tbody>
    </table>
    <span>
      <img style={styles.editInfoPencil} src="https://png.icons8.com/ios/50/2e6d97/edit.png" />
      <a href="https://www.yelp.com" style={styles.editInfoStyling}>
          Edit business info
      </a>
    </span>
  </div>
);

export default Hours;
