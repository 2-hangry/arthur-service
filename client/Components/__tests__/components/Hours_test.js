import React from 'react';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import Hours from '../../HoursOfOperation/Hours.jsx';

Enzyme.configure({ adapter: new Adapter() });

it('renders correctly', () => {
  const props = {
    hours: {
      open: [
        {
          day: 0,
          end: 2014,
          start: 897,
        },
        {
          day: 1,
          end: 2100,
          start: 700,
        },
        {
          day: 2,
          end: 2300,
          start: 700,
        },
        {
          day: 3,
          end: 1600,
          start: 300,
        },
        {
          day: 4,
          end: 2100,
          start: 700,
        },
        {
          day: 5,
          end: 2300,
          start: 700,
        },
        {
          day: 6,
          end: 1600,
          start: 300,
        },
      ],
    },
  };

  const tree = renderer.create(<Hours {...props} />).toJSON();
  expect(tree).toMatchSnapshot();
});
