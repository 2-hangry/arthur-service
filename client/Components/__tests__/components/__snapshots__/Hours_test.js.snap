// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <h3
    style={
      Object {
        "color": "red",
        "fontSize": 14,
      }
    }
  >
    Hours
  </h3>
  <table>
    <tbody>
      <tr
        style={
          Object {
            "fontFamily": "sans-serif",
            "fontSize": "14px",
            "textAlign": "left",
          }
        }
      >
        <th>
          Mon
        </th>
        <td>
          08:00 am - 08:00 pm
        </td>
        <p />
      </tr>
      <tr
        style={
          Object {
            "fontFamily": "sans-serif",
            "fontSize": "14px",
            "textAlign": "left",
          }
        }
      >
        <th>
          Tue
        </th>
        <td>
          07:00 am - 09:00 pm
        </td>
        <p />
      </tr>
      <tr
        style={
          Object {
            "fontFamily": "sans-serif",
            "fontSize": "14px",
            "textAlign": "left",
          }
        }
      >
        <th>
          Wed
        </th>
        <td>
          07:00 am - 11:00 pm
        </td>
        <p />
      </tr>
      <tr
        style={
          Object {
            "fontFamily": "sans-serif",
            "fontSize": "14px",
            "textAlign": "left",
          }
        }
      >
        <th>
          Thu
        </th>
        <td>
          03:00 am - 04:00 pm
        </td>
        <p />
      </tr>
      <tr
        style={
          Object {
            "fontFamily": "sans-serif",
            "fontSize": "14px",
            "textAlign": "left",
          }
        }
      >
        <th>
          Fri
        </th>
        <td>
          07:00 am - 09:00 pm
        </td>
        <p />
      </tr>
      <tr
        style={
          Object {
            "fontFamily": "sans-serif",
            "fontSize": "14px",
            "textAlign": "left",
          }
        }
      >
        <th>
          Sat
        </th>
        <td>
          07:00 am - 11:00 pm
        </td>
        <p
          style={
            Object {
              "color": "green",
              "fontWeight": "bold",
            }
          }
        >
          Open Now
        </p>
      </tr>
      <tr
        style={
          Object {
            "fontFamily": "sans-serif",
            "fontSize": "14px",
            "textAlign": "left",
          }
        }
      >
        <th>
          Sun
        </th>
        <td>
          03:00 am - 04:00 pm
        </td>
        <p />
      </tr>
    </tbody>
  </table>
</div>
`;
